src/engine/controls/cuecontrol.cpp

```diff
@@ -432,27 +440,33 @@ void CueControl::trackLoaded(TrackPointer pNewTrack) {
         // This prevents jumps when track analysis finishes while quantization is enabled.
         double cuePoint = m_pCuePoint->get();
         if (cuePoint != Cue::kNoPosition) {
-            seekExact(cuePoint);
+            seekOnLoad(cuePoint);
         } else {
-            seekExact(0.0);
+            seekOnLoad(0.0);
         }
         break;
     }
     case SeekOnLoadMode::IntroStart: {
         double introStart = m_pIntroStartPosition->get();
         if (introStart != Cue::kNoPosition) {
-            seekExact(introStart);
+            seekOnLoad(introStart);
         } else {
-            seekExact(0.0);
+            seekOnLoad(0.0);
         }
         break;
     }
     default:
-        seekExact(0.0);
+        DEBUG_ASSERT(!"Unknown enum value");
+        seekOnLoad(0.0);
         break;
     }
 }
 
+void CueControl::seekOnLoad(double seekOnLoadPosition) {
+    seekExact(seekOnLoadPosition);
+    m_usedSeekOnLoadPosition.setValue(seekOnLoadPosition);
```

Comment:
Why not `m_lastSeekOnLoadPosition`?

Reply:
Because it is only relevant if the last seek position has been used. 
The last position is not used in case of vinyl control.